dht.go

```diff
@@ -117,6 +128,8 @@ func New(ctx context.Context, h host.Host, options ...opts.Option) (*IpfsDHT, er
 	// register for network notifs.
 	dht.host.Network().Notify((*netNotifiee)(dht))
 
+	go dht.handleProtocolChanges(ctx)
```

Comment:
what are we listening for? The subscription to the eventbus takes place in the goroutine.

Reply:
Conversation here: https://github.com/libp2p/go-libp2p-kad-dht/pull/472#discussion_r387423621

Reply:
@aschmahmann 

Let's use a go-process here.

Reply:
Also, let's move the subscription here as even if you walk the peerstore as described in #472 , you could still miss peers that start listening to the DHT protocol after that but before the subscription in "handleProtocol" changes gets created. 

IMO, it's better than walking the store again for this subscription.

Reply:
@aschmahmann Let's use a go- process here.

Reply:
Also, please note that even after #472 lands, you would ideally have to walk the peer store again after this subscription is made because:

1. We are connected to a peer that dosen't support the DHT protocol before we start the DHT.
2. The Notfiee in #472 walks, subscribes etc but ofcourse discards the peer in 1.
3. The peer starts supporting the DHT protocol.
4. We now kickoff "handleProtocolChanges" thus missing the peer.

Reply:
> Let's use a go- process here.

Just for consistency or anything else? I'd kind of like to just remove goprocess from this library entirely. Switching the query logic to using normal goroutines and contexts made life much easier.

We can do a goprocess purge in a future PR though

Reply:
> you would ideally have to walk the peer store again after this subscription is made

I actually would like to just bundle these together into the same goroutine. We're going to be selecting on channels with low throughput, why not bundle them up. We can then do:
1) Subscribe for IdentifyCompleted
2) Subscribe for ProtocolsUpdated
3) Walk peerstore
4) Listen for, and process, events